src/api/firebase.js

```diff
@@ -72,10 +73,67 @@ export async function addItem(listId, { itemName, daysUntilNextPurchase }) {
 	});
 }
 
-export async function updateItem(listId, itemId, currentItemUpdates) {
+export async function updateItem(
+	listId,
+	itemId,
+	dateLastPurchased,
+	dateNextPurchased,
+	dateCreated,
+	currentItemUpdates,
+) {
 	const currentItemRef = doc(db, listId, itemId);
+	// use updated totalPurchases
+	const { totalPurchases } = currentItemUpdates;
+
+	let daysSinceLastTransaction;
+	let previousEstimate;
+	// When the item was purchased first time
+	if (totalPurchases === 1) {
+		// number of days from the option selected by the user
+		daysSinceLastTransaction = getDaysBetweenDates(
+			dateCreated.toDate(),
+			dateNextPurchased.toDate(),
```

Comment:
**Question:** For when totalPurchases === 1, should daysSinceLastTransaction be 
`getDaysBetweenDates( dateCreated.toDate(), dateLastPurchased.toDate() );` ? I may be wrong but when I think of the days between when I add an item to my list and when I'm checking off that item to say I want to purchase it, I would think the difference of days would be when I added it and when I set to actually buying it the first time.

Previously when I added Mango to my list, the dateNextPurchased was Sep 18th but after checking it off, it changed to Sep 22nd. Shouldn't daysSinceLastTransaction be 5 in this case instead of 7? (Sep 11th -> Sep 15th vs. Sep 11th -> Sep 18th)
![Screen Shot 2023-09-15 at 1 54 11 PM](https://github.com/the-collab-lab/tcl-64-smart-shopping-list/assets/113865981/74c4d491-a8d2-4e62-90c3-0bb43215eb6b)
![Screen Shot 2023-09-15 at 1 56 38 PM](https://github.com/the-collab-lab/tcl-64-smart-shopping-list/assets/113865981/157f7100-7d76-4a77-9bca-844d35f55a14)

Reply:
Good catch @ismarjiw. The way you've explained it here makes more sense to me, too. Especially considering this comment within the `calculateEstimate` function definition:

`daysSinceLastTransaction, // The number of days since the item was added to the list or last purchased`

with '**_since the item was added to the list_**' being the key words for this discussion.

Reply:
You read the task right, Ismarji and Tiala. I think this is what the creator wants here. I will update the code.

One thing that doesn't make sense is that we throw away the user's selection(soon, kind of soon, not soon). Probably, it's just used for sorting on the next task. I was thinking too much.